h2o-automl/src/test/java/ai/h2o/automl/AutoMLTest.java

```diff
@@ -34,4 +36,29 @@
       if(fr != null) fr.remove();
     }
   }
+
+  @Test public void KeepCrossValidationFoldAssignmentTest() {
+    AutoML aml=null;
+    Frame fr=null;
+    try {
+      AutoMLBuildSpec autoMLBuildSpec = new AutoMLBuildSpec();
+      fr = parse_test_file("./smalldata/airlines/allyears2k_headers.zip");
+      autoMLBuildSpec.input_spec.training_frame = fr._key;
+      autoMLBuildSpec.input_spec.response_column = "IsDepDelayed";
+      autoMLBuildSpec.build_control.stopping_criteria.set_max_runtime_secs(5);
+      autoMLBuildSpec.build_control.keep_cross_validation_fold_assignment = true;
+
+      aml = AutoML.makeAutoML(Key.<AutoML>make(), new Date(), autoMLBuildSpec);
+      AutoML.startAutoML(aml);
+      aml.get();
+
+      assertTrue(aml.leader() !=null && aml.leader()._parms._keep_cross_validation_fold_assignment);
+      assertTrue(aml.leader() !=null && aml.leader()._output._cross_validation_fold_assignment_frame_id != null);
```

Comment:
I think this will only work if the leader is not a Stacked Ensemble.  Did you try running the test manually to see what happens?

Reply:
This should work fine. SE is just a normal Model in h2o.

Reply:
Actually I wanted to ask whether we are writing Java tests or most of the times just R and Python tests? I'm judging by the number of java tests in h2o-automl module.

Reply:
@deil87 We should have some JUnits, imho. They can help catch other things like leaked keys, which basically  means AutoML is not cleaning up well.

Reply:
@ledell probably you are right... at least in R tests I was not able to find fold assignments in SE model. Test were ok when I was running them locally... but it could be just lucky coincidence